crates/krator/src/manager/controller.rs

```diff
@@ -0,0 +1,197 @@
+use super::watch::{Watch, WatchHandle};
+#[cfg(feature = "admission-webhook")]
+use crate::admission::WebhookFn;
+use crate::Operator;
+use kube::api::ListParams;
+use kube::Resource;
+
+/// Builder pattern for registering a controller or operator.
+pub struct ControllerBuilder<C: Operator> {
+    /// The controller or operator singleton.
+    pub(crate) controller: C,
+    ///  List of watch configurations for objects that will simply be cached
+    ///  locally.
+    pub(crate) watches: Vec<Watch>,
+    /// List of watch configurations for objects that will trigger
+    /// notifications (based on OwnerReferences).
+    pub(crate) owns: Vec<Watch>,
+    /// Restrict our controller to act on a specific namespace.
+    namespace: Option<String>,
+    /// Restrict our controller to act on objects that match specific list
+    /// params.
+    list_params: ListParams,
+}
+
+/// Trait alias for types which can be watched.
+pub trait Watchable:
+    Resource<DynamicType = ()> + serde::de::DeserializeOwned + Clone + Send + 'static
+{
+}
+
+impl<T> Watchable for T where
+    T: Resource<DynamicType = ()> + serde::de::DeserializeOwned + Clone + Send + 'static
+{
+}
+
+impl<O: Operator> ControllerBuilder<O> {
+    /// Create builder from operator singleton.
+    pub fn new(operator: O) -> Self {
+        ControllerBuilder {
+            controller: operator,
+            watches: vec![],
```

Comment:
So, I think this requires `Default` to be implemented for `ControllerBuilder`, which I don't think is easy because of the `controller` field (certainly can't derive). Am I missing something?

Reply:
Oh right, because it is calling the parent default method.